pkg/kubelet/kubelet.go

```diff
@@ -1636,7 +1636,13 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 		// Introduce some small jittering to ensure that over time the requests won't start
 		// accumulating at approximately the same time from the set of nodes due to priority and
 		// fairness effect.
-		go wait.JitterUntil(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, 0.04, true, wait.NeverStop)
+		go func() {
+			// Call updateRuntimeUp once before syncNodeStatus to make sure kubelet had already checked runtime state
+			// otherwise when restart kubelet, syncNodeStatus will report node notReady in first report period
+			kl.updateRuntimeUp()
+			wait.JitterUntil(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, 0.04, true, wait.NeverStop)
+		}()
+
 		go kl.fastStatusUpdateOnce()
```

Comment:
since we are doing this I prefer to put the goroutines in the right order https://github.com/kubernetes/kubernetes/issues/124397#issuecomment-2071700205

Reply:
@AllenXu93 please address above comment, thanks.

Reply:
There has been a lot of discussion on this issue in #124397, @aojea do you think the current behavior is reasonable or do we need to discuss it further?

Reply:
I'm fine with current solution and agree with the points expressed there, blocking on the updateRuntimeUp operation is the only safe way ... thanks for correct me